overlord/hookstate/hooks.go

```diff
@@ -234,9 +233,17 @@ func (h *gateAutoRefreshHookHandler) Error(hookErr error) (ignoreHookErr bool, e
 	// the hook didn't request --hold, or it was --proceed. since the hook
 	// errored out, assume hold.
 
-	var affecting []string
-	if err := ctx.Get("affecting-snaps", &affecting); err != nil {
-		return false, fmt.Errorf("internal error: cannot get affecting-snaps")
+	affected, err := snapstate.AffectedByRefreshCandidates(st)
+	if err != nil {
+		return false, err
+	}
+	affectedInfo := affected[snapName]
+	if affectedInfo == nil || len(affectedInfo.AffectingSnaps) == 0 {
+		return false, nil
+	}
+	affecting := make([]string, 0, len(affectedInfo.AffectingSnaps))
+	for sn := range affectedInfo.AffectingSnaps {
+		affecting = append(affecting, sn)
```

Comment:
and this code look quite similar, wondering if there's a helper that could be defined and shared about this

Reply:
Good idea, done.